src/components/structures/MessagePanel.js

```diff
@@ -296,6 +296,12 @@ module.exports = React.createClass({
             // Wrap consecutive member events in a ListSummary
             if (isMembershipChange(mxEv)) {
                 let ts1 = mxEv.getTs();
+                // Ensure that the key of the MemberEventListSummary does not change with new
+                // member events. This will prevent it from being re-created necessarily, and
+                // instead will allow new props to be provided. In turn, the shouldComponentUpdate
+                // method on MELS can be used to prevent unnecessary renderings.
+                // `prevEvent` at this point is a non-member event or null.
+                const key = "memberlistsummary-" + (prevEvent ? prevEvent.getId() : "");
```

Comment:
> How about (prevEvent ? mxEv.getId() : 'initial');

This misses the point. When back paginating a lot of member events, `mxEv` is constantly changing as it represent the first/earliest member event. If you key off the event ID for `mxEv` it will be constantly changing as well, leading to unnecessary teardown/setting up of new `MemberEventListSummary` classes, rather than passing them through as prop updates which is what you really want.